server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -902,6 +909,23 @@ public static String findV2Template(Metadata metadata, String indexName, boolean
         Optional.ofNullable(template.template())
             .map(Template::mappings)
             .ifPresent(mappings::add);
+
+        // Only include _timestamp mapping snippet if creating backing index.
+        if (indexName.startsWith(DataStream.BACKING_INDEX_PREFIX)) {
```

Comment:
@dakrone I had to add this if statement otherwise the meta field mapper was going to be applied on each create index api call. In a docs test, the new logs-*-* composable index template was triggered by a regular create index api call and then the test failed, because the document being indexed had no timestamp field. 

I think only applying the meta field automatically makes sense for backing indices only and not when a user creates a new index via create index api and the index composable template matches?

See commits: a23d2e484594b5cbe788f3ef4bfcd40f97136f67, which then was superseded by: c78733204c12cf44b53026052aff9865d99ced03

Reply:
Note currently in master, the `_timestamp` field gets only applied if the create index request originates from create data stream api and data stream rollover (signalled via `CreateIndexClusterStateUpdateRequest#dataStreamName` field), but now this logic moved `MetadataIndexTemplateService.collectMappings(...)` method, which didn't have that context.

Reply:
Would it be cleaner to pull this check out of `collectMappings` and into the caller? That way we could avoid the workaround here: https://github.com/elastic/elasticsearch/pull/59076/files#diff-115cd2f0fff9dd97acc95e1d29012a15R1075.